source/api_cc/src/DeepPot.cc

```diff
@@ -1234,30 +1234,17 @@ void DeepPot::get_type_map(std::string& type_map) {
   type_map = get_scalar<STRINGTYPE>("model_attr/tmap");
 }
 
-DeepPotModelDevi::DeepPotModelDevi()
-    : inited(false), init_nbor(false), numb_models(0) {}
+DeepPotModelDevi::DeepPotModelDevi() : inited(false), numb_models(0) {}
 
 DeepPotModelDevi::DeepPotModelDevi(
     const std::vector<std::string>& models,
     const int& gpu_rank,
     const std::vector<std::string>& file_contents)
-    : inited(false), init_nbor(false), numb_models(0) {
-  try {
-    init(models, gpu_rank, file_contents);
-  } catch (...) {
-    // Clean up and rethrow, as the destructor will not be called
-    for (unsigned ii = 0; ii < numb_models; ++ii) {
-      delete graph_defs[ii];
-    }
-    throw;
-  }
+    : inited(false), numb_models(0) {
+  init(models, gpu_rank, file_contents);
```

Comment:
Consider handling exceptions in the `DeepPotModelDevi` constructor where `init` function is called. This can help in managing any potential errors during the initialization of the class. [important]